repos/spack_repo/builtin/packages/adios2/package.py

```diff
@@ -252,6 +252,10 @@ class Adios2(CMakePackage, CudaPackage, ROCmPackage):
     # https://github.com/ornladios/ADIOS2/pull/4214
     patch("2.10-enable-rocm6.patch", when="@2.9.1:2.10.1")
 
+    # Fix issue with GCC 7
+    # https://github.com/ornladios/ADIOS2/pull/4591
+    patch("2.10-gcc-7-compatability", when="@2.8:2.10")
+
```

Comment:
```suggestion
    # Fix issue with GCC 7
    patch(
        "https://github.com/ornladios/adios2/commit/b7a5957.patch?full_index=1",
        sha256="d854008ab27d6ebfa66fffb78126b17713cda3234ed19bf331f85a720e599a32",
        when="@2.8:2.10",
    )

```

Whenever possible we should get out of the habit of putting patches directly into the packages repository. Instead it's better to reference the merge commit from GitHub.